sympy/matrices/dense.py

```diff
@@ -880,7 +880,12 @@ def row_del(self, i):
         row
         col_del
         """
-        self._mat = self._mat[:i*self.cols] + self._mat[(i + 1)*self.cols:]
+        if i >= -self.rows and i < self.rows:
+            self._mat = self._mat[:i*self.cols] + self._mat[(i + 1)*self.cols:]
+        elif i < -self.rows:
+            self._mat = self._mat[self.cols:]
```

Comment:
> Maybe this case should raise IndexError(...

probably it's best to do that. That would change the current behavior, however, whereas the current changes make negative indices that are out of range behave the same as positive indices. That's why I am +1 to the current change; raising an error can be a new PR.

But it would also be nice to see `col_del` fixed with this PR.

```python
>>> i
Matrix([[1, 0, 0]])
>>> i.col_del(-11)
>>> i
Matrix([[1, 0]])  <--- the 1 should have been removed
```